pkg/repository/git.go

```diff
@@ -73,7 +73,7 @@ func (r gitRepoController) isClean(ctx context.Context) (bool, error) {
 	timeoutErr := errors.New("timeout exceeded")
 
 	go func() {
-		time.Sleep(time.Second)
+		time.Sleep(5 * time.Second)
```

Comment:
Reviewing the flow, this might be called at least 3 times in a `deploy` sequence. This could have an impact of 15 seconds on those deployments and we already had performance issues because of it. I would avoid to increase this timeout

Reply:
I agree with @ifbyol, the timeout should be 1s, otherwise it defeats the purpose of being there. Once we refactor the code and avoid calling this function many times, I agree in bumping up the timeout to 5s

Reply:
That change has not been scheduled in out sprint yet, I will talk to @rlamana about it